guava/src/com/google/common/collect/CartesianList.java

```diff
@@ -70,6 +70,27 @@ private int getAxisIndexForProductIndex(int index, int axis) {
     return (index / axesSizeProduct[axis + 1]) % axes.get(axis).size();
   }
 
+  @Override
+  public int indexOf(Object o) {
+    if (!(o instanceof List)) {
+      return -1;
+    }
+    List<?> list = (List<?>) o;
+    if (list.size() != axes.size()) {
+      return -1;
+    }
+    ListIterator<?> itr = list.listIterator();
+    int computedIndex = 0;
+    while (itr.hasNext()) {
+      int axeIndex = itr.nextIndex(), elemIndex;
```

Comment:
Google Style doesn't allow for multiple variables to be declared in the same statement. Can you split this up?

I think it would also make the next conditional much clearer:

```java
int axisIndex = itr.nextIndex();
int elemIndex = axes.get(axisIndex).indexOf(itr.next());
if (elemIndex == -1) {
  return -1;
}
...
```